### PR TITLE
Initial type definitions for @mailchimp/mailchimp_marketing

### PR DESCRIPTION
**Pull request by rattkin**
Adding a new definition:
- [x ] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [ x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "dtslint/dt.json" }`, and no additional rules.
- [ x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/mailchimp__mailchimp_marketing/mailchimp__mailchimp_marketing-tests.ts [hunk 1]
```diff
@@ -0,0 +1,3 @@
+// test: Usage of UMD Globals should be prohibited inside modules
+
+export {}; // dummy export to mark this file as a module
```
Comment (peterblazejewicz) [inline types/mailchimp__mailchimp_marketing/mailchimp__mailchimp_marketing-tests.ts:3]: this is for declaration file usage, while this is proper `.ts` file.
This file should test your declarations

**Comment by peterblazejewicz** [comment]
is this for a native JS package from here?
https://unpkg.com/browse/@mailchimp/mailchimp_marketing@3.0.45/src/

**Comment by rattkin** [comment]
> is this for a native JS package from here?
> https://unpkg.com/browse/@mailchimp/mailchimp_marketing@3.0.45/src/

it's for this: https://www.npmjs.com/package/@mailchimp/mailchimp_marketing

Please note that while NPM autodetects it ships type definitions in the package, in reality it does not. I reported this issue https://github.com/mailchimp/mailchimp-client-lib-codegen/issues/134

**Comment by orta** [comment]
I commented on the mailchimp original issue - https://github.com/mailchimp/mailchimp-client-lib-codegen/issues/134

Basically, I'd like to give Mailchimp a polite poke to take a look - otherwise I think another DT reviewer should consider merging it

**Review by orta — APPROVED** [review]
No response from a mailchimp contributor - so I think this is probably reasonable to go

**Comment by rattkin** [comment]
Ready to merge

**Pull request merged**